### PR TITLE
First implementation of ActivityPub C2S

### PR DESCRIPTION
**Pull request by annando**
This PR adds a very basic and experimental support for "ActivityPub Client to Server", means the part between a client software and the system. The only client that I currently know of is AndStatus. And at least some data is visible, although the system behaves weird, see https://github.com/andstatus/andstatus/issues/573

Posting is currently not supported and it is far from "Production Status".

### CODE REVIEW

File: src/Module/ActivityPub/Whoami.php [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+
+use Friendica\Content\Text\BBCode;
+use Friendica\Core\System;
+use Friendica\DI;
+use Friendica\Model\User;
+use Friendica\Module\BaseApi;
+use Friendica\Protocol\ActivityPub;
+
+/**
+ * Dummy class for all currently unimplemented endpoints
```
Comment (MrPetovan) [inline src/Module/ActivityPub/Whoami.php]: Is this accurate?

File: src/Object/Api/Mastodon/Token.php [hunk 2]
```diff
@@ -46,12 +48,30 @@ class Token extends BaseDataTransferObject
 	 * @param string $token_type
 	 * @param string $scope
 	 * @param string $created_at
+	 * @param string $me
```
Comment (MrPetovan) [inline src/Object/Api/Mastodon/Token.php]: Please explicit the expected format of this parameter (a profile URL?).

**Review by MrPetovan — CHANGES_REQUESTED** [review]

File: src/Module/ActivityPub/Whoami.php (hunk 1 above)
Comment (annando) [reply src/Module/ActivityPub/Whoami.php]: Mostly. :grin:

Comment (annando) [reply src/Module/ActivityPub/Whoami.php]: Done

File: src/Object/Api/Mastodon/Token.php (hunk 2 above)
Comment (MrPetovan) [reply src/Object/Api/Mastodon/Token.php]: Please add parameter documentation so that I can merge this PR.

**Review by MrPetovan — APPROVED** [review]

**Pull request merged**